src/ratgdo-platform.ts

```diff
@@ -335,10 +336,10 @@ export class RatgdoPlatform implements DynamicPlatformPlugin {
     const device = {
 
       address: address,
-      firmwareVersion: deviceInfo.version,
+      firmwareVersion: deviceInfo.version || deviceInfo.esphome_version,
```

Comment:
This should use the nullish-coalescing operator `??` instead. It's more appropriate here.